src/basic/path-util.c

```diff
@@ -772,6 +772,19 @@ bool hidden_or_backup_file(const char *filename) {
         if (!p)
                 return false;
 
+        /* Please, let's not add more entries to the list below. If external projects think it's a good idea to come up
+         * with always new suffixes and that everybody else should just adjust to that, then it really should be on
+         * them. Hence, in future, let's not add any more entries. Instead, let's ask those packages to instead adopt
+         * one of the generic suffixes/prefixes for hidden files or backups, possibly augmented with an additional
+         * string. Specifically: there's now:
+         *
+         *    The generic suffixes "~" and ".bak" for backup files
+         *    The generic prefix "." for hidden files
+         *
+         * Thus, if a new package manager "foopkg" wants its own set of "foopkg-new", "foopkg-old", "foopkg-dist" or so
+         * registered, let's refuse that and ask them to use "foopkg-new.bak" or "foopkg-new~" instead.
```

Comment:
Agreed, thanks. The rpm/dpkg/ucf stuff much predates concepts like `sysv-generator`, so I guess we have to grandfather them, but this is indeed a nuisance and too much guesswork.